src/backend/distributed/utils/citus_clauses.c

```diff
@@ -141,12 +146,39 @@ ExecuteMasterEvaluableFunctions(Query *query, PlanState *planState)
 	{
 		RangeTblEntry *rte = (RangeTblEntry *) lfirst(rteCell);
 
-		if (rte->rtekind != RTE_SUBQUERY)
+		if (rte->rtekind == RTE_SUBQUERY)
 		{
-			continue;
+			ExecuteMasterEvaluableFunctions(rte->subquery, planState);
+		}
+		else if (rte->rtekind == RTE_VALUES)
+		{
+			ListCell *exprListCell = NULL;
+
+			foreach(exprListCell, rte->values_lists)
+			{
+				List *exprList = (List *) lfirst(exprListCell);
+				ListCell *exprCell = NULL;
+
+				foreach(exprCell, exprList)
+				{
+					Expr *expr = (Expr *) lfirst(exprCell);
+					Node *modifiedNode = NULL;
+
+					if (commandType == CMD_INSERT && !insertSelectQuery)
+					{
+						modifiedNode = EvaluateNodeIfReferencesFunction((Node *) expr,
```

Comment:
Small note: we'll probably stop special casing INSERTs here (#1506). It seems to be a performance hack, which actually causes some bugs / limitations.